teuthology/task/internal.py

```diff
@@ -393,6 +393,35 @@ def archive(ctx, config):
                 path = os.path.join(logdir, rem.shortname)
                 misc.pull_directory(rem, archive_dir, path)
 
+                #Check for Coredumps:
+                coredump_path = os.path.join(path, 'coredump')
+                if os.path.isdir(coredump_path):
+                    log.info('Transferring binaries for coredumps...')
+                    for dump in os.listdir(coredump_path):
+                        # Pull program from core file
+                        dump_path = os.path.join(coredump_path, dump)
+                        dump_info = subprocess.Popen(['file', dump_path], stdout=subprocess.PIPE)
+                        dump_out = dump_info.communicate()
+                        dump_program = dump_out.split("from '")[1].split(' ')[0]
```

Comment:
@zmc What do you think zack?

I think using split is probably slightly faster than re.match() and doesn't require doing an import re,  and actually easier to read (to me) but... I usually have to think about it a little when it comes to regular expressions. I am fine either way. I use re.match() quite a bit too depending on the situation.